src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,185 @@
+                connection.prepareStatement("INSERT INTO orders (user_id) VALUES(?)",
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                order.setOrderId(generatedKeys.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create Order with id: "
+                    + order.getOrderId(), e);
+        }
+        for (Product product : order.getProducts()) {
+            addDataToOrdersProductTable(order.getOrderId(), product.getProductId());
+        }
+        return order;
+    }
+
+    @Override
+    public Optional<Order> getById(Long id) {
+        Order order = null;
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement("SELECT order_id, user_id FROM orders "
+                        + "WHERE order_id = ? AND is_deleted = FALSE ")) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = new Order(resultSet.getLong("order_id"),
+                        resultSet.getLong("user_id"));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Order with id: " + id, e);
+        }
+        order.setProducts(getOrdersProducts(order.getOrderId()));
+        return Optional.of(order);
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "UPDATE orders SET user_id = ? "
+                + "WHERE order_id = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, order.getUserId());
+            preparedStatement.setLong(2, order.getOrderId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order with id "
+                    + order.getOrderId(), e);
+        }
+        deleteOrderProducts(order.getOrderId());
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement("UPDATE orders "
+                        + "SET is_deleted = TRUE WHERE order_id = ?")) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Order with id: " + id, e);
+        }
+        return deleteOrderProducts(id);
```

Comment:
Instead of this, you can do this before the catch block:
`return statement.executeUpdate() > 0;`